File: android/src/main/kotlin/com/pravera/flutter_foreground_task/models/NotificationButton.kt
package com.pravera.flutter_foreground_task.models

import org.json.JSONObject
import java.util.Objects

data class NotificationButton(
    val id: String,
    val text: String,
    val textColorRgb: String?
) {
    companion object {
        private const val ID_KEY = "id"
        private const val TEXT_KEY = "text"
        private const val TEXT_COLOR_RGB_KEY = "textColorRgb"

        fun fromJSONObject(jsonObj: JSONObject): NotificationButton {
            val id: String = if (jsonObj.isNull(ID_KEY)) {
                "unknown"
            } else {
                jsonObj.getString(ID_KEY)
            }

            val text: String = if (jsonObj.isNull(TEXT_KEY)) {
                ""
            } else {
                jsonObj.getString(TEXT_KEY)
            }

            val textColorRgb: String? = if (jsonObj.isNull(TEXT_COLOR_RGB_KEY)) {
                null
            } else {
                jsonObj.getString(TEXT_COLOR_RGB_KEY)
            }

            return NotificationButton(id = id, text = text, textColorRgb = textColorRgb)
        }
    }

    override fun equals(other: Any?): Boolean {
        if (other == null || other !is NotificationButton) {
            return false
        }
        return this.id == other.id && this.text == other.text && this.textColorRgb == other.textColorRgb
    }

    override fun hashCode(): Int {
        return Objects.hash(id, text, textColorRgb)
    }
}

File: ios/flutter_foreground_task/Sources/flutter_foreground_task/models/NotificationButton.swift
//
//  NotificationButton.swift
//  flutter_foreground_task
//
//  Created by Woo Jin Hwang on 8/5/24.
//

import Foundation

private let ID_KEY = "id"
private let TEXT_KEY = "text"

struct NotificationButton {
  let id: String
  let text: String
  
  static func fromJSONObject(_ jsonObj: Any) -> NotificationButton {
    var id: String = "unknown"
    var text: String = ""
    
    if let _jsonObj = jsonObj as? Dictionary<String, Any> {
      if let _id = _jsonObj[ID_KEY] as? String {
        id = _id
      }
      
      if let _text = _jsonObj[TEXT_KEY] as? String {
        text = _text
      }
    }
    
    return NotificationButton(id: id, text: text)
  }
}
